src/components/Experience.ts

```diff
@@ -29,11 +30,85 @@ export class Experience<TGameStartr extends FullScreenPokemon> extends GeneralCo
             this.levelup(pokemon);
+            return true;
+        }
+        return false;
+    }
+
+     /**
+      * Processes experience gain for one or more pokemon
+      *
+      * @param battleInfo   Information about the current battle.
+      * @param onComplete   Handler for when this is done.
+      */
+    public processBattleExperience(battleInfo: IBattleInfo, onComplete: () => void): void {
+        const experienceToGain =  this.gameStarter.equations.experienceGained(
+            battleInfo.teams.player, battleInfo.teams.opponent);
+
+        const experienceText: (string | string[])[][] = [[
+            battleInfo.teams.player.selectedActor.nickname,
+            ` gained ${experienceToGain} experience!`]];
+        this.gameStarter.menuGrapher.createMenu("GeneralText");
+        const levelUp = this.gainExperience(battleInfo.teams.player.selectedActor, experienceToGain);
+        if (levelUp) {
+            this.gameStarter.menuGrapher.addMenuDialog(
+                "GeneralText", experienceText,
+                () => this.processBattleLevelUp(battleInfo.teams.player.selectedActor, onComplete));
+        } else {
+            this.gameStarter.menuGrapher.addMenuDialog("GeneralText", experienceText, onComplete);
         }
+
+        this.gameStarter.menuGrapher.setActiveMenu("GeneralText");
     }
+
+     /**
+      * Processes level up for a given pokemon
+      *
+      * @param pokemon   Pokemon who is going to level up.
+      * @param onComplete   Handler for when this is done.
+      */
+    private processBattleLevelUp(pokemon: IPokemon, onComplete: () => void) {
+        this.gameStarter.menuGrapher.createMenu("GeneralText");
+        const experienceText: (string | string[])[][] = [[
+            pokemon.nickname,
+            ` grew to level ${pokemon.level}!`]];
+        if (this.canLearnMoveAtLevel(pokemon)) {
+            this.gameStarter.menuGrapher.addMenuDialog(
+                "GeneralText", experienceText,
+                () => this.learnBattleMove(pokemon, onComplete));
+        } else {
+            this.gameStarter.menuGrapher.addMenuDialog(
+                "GeneralText", experienceText, onComplete);
+        }
+
+        this.gameStarter.menuGrapher.setActiveMenu("GeneralText");
+    }
+
+     /**
+      * Determines whether a Pokemon should learn a move
+      *
+      * @param pokemon   Pokemon who is going to going to learn a move.
+      * @returns Whether a Pokemon can learn a move at its current level
+      */
+    private canLearnMoveAtLevel(pokemon: IPokemon): boolean {
+        return false;
+    }
+
+     /**
+      * Processes learning a move during a battle for a given pokemon
+      *
+      * @param pokemon   Pokemon who is going to going to learn a move.
+      * @param onComplete   Handler for when this is done.
+      *
+      */
+    private learnBattleMove(pokemon: IPokemon, onComplete: () => void) {
```

Comment:
You got rid of the return, but neglected to call `onComplete` :)

Reply:
Also, have you verified this whole thing actually works? It looks like because this never calls `onComplete` it wouldn't?